docs/terraform-generation.md

```diff
@@ -0,0 +1,115 @@
+# Terraform Code Generation
+
+Terramate provides a way to integrate its [globals](globals.md) and
```

Comment:
This sentence is a bit confusing, maybe because it says `Terramate ... Terraform ... Terramate ...`.

What about:

```
The tool supports the generation of arbitrary Terraform code using 
both [globals](globals.md) and [metadata](metadata.md) from Terramate.
```

but for me reading this, it looks like `globals` and `metadata` is already implicit in the Terramate context, so maybe this is enough:

```
Terramate supports the generation of arbitrary Terraform code using 
both [globals](globals.md) and [metadata](metadata.md).
```